### DISCUSSION TITLE
Left Side key not responding

### DISCUSSION THREAD

**Discussion by ricelovinbear** [Yowkees/keyball#482] [Q&A] [answered]
![KakaoTalk_20240212_121142809](https://github.com/Yowkees/keyball/assets/159744624/76bc14a7-4fc2-4fb3-a55d-de1257ccc444)
![KakaoTalk_20240212_132625323](https://github.com/Yowkees/keyball/assets/159744624/c88a76ef-deca-4a43-839d-86f365dcdc0c)
![KakaoTalk_20240212_132625323_01](https://github.com/Yowkees/keyball/assets/159744624/2f0b36db-3301-434e-bb59-6dd7d1eb893a)

Hello I have recently purchased Keyball 39 and got it soldered. 
LED and OLED screen works fins TRRS fine too but left side keys don't respond on it's own and right side works on it's own. 
But when I attach TRRS cable right side recognize as left?? Can I get any help on this?

こんにちは！最近keyball39を購入してはんだ付けはおわりました。
LEDと画面は問題なく機能しますが左側のキーの反応がないです。別々につなげたら左は反応なしで右は反応ありです。
TRRSでつなげると左は反応ぜす右側が左に認識されます。
どこが問題かアドバイスお願いできますでしょうか？

**Comment by Ryo624** [comment, marked as answer]
Thank you for contacting us.

It appears that the solder on the terminal at the red circle in the photo has touched the adjacent terminal.
Is it possible to avoid touching adjacent terminals?

Also try replacing the TRS cable
![303970870-c88a76ef-deca-4a43-839d-86f365dcdc0c](https://github.com/Yowkees/keyball/assets/143698349/d9fdcdff-2049-4dce-97d6-f26b1419f5a8)

**Reply by ricelovinbear** [reply]
thank you for the reply. I will try to check ASAP. I'm going to bring it to the place where I got the soldering service and give that a try.
As for the TRS cable what do you recommend?

**Reply by Ryo624** [reply]
I want to see if changing to a different cable will fix it, so any cable will do.

**Reply by ricelovinbear** [reply]
bought new cable but have same problem
going to try to bring the product tomorrow for resoldering

**Reply by Ryo624** [reply]
One more thing I would like you to try: what are the symptoms if you swap the ProMicro from left to right?

**Reply by ricelovinbear** [reply]
just tried switching the ProMicro
when line is connected to the left ProMicro, only right side works with leftside keymapping but when line is connected to the right ProMicro, only right works with correct keymapping(righthand keymapping)

**Reply by ricelovinbear** [reply]
![20240306_133333](https://github.com/Yowkees/keyball/assets/159744624/318cf65d-997f-4da2-b570-c3250e1fcce4)

is this OK?

**Reply by Ryo624** [reply]
Is that with the TRRS cable connected?

Soldering is okay.

**Reply by ricelovinbear** [reply]
Yes with the cable connected. 
Could it be something wrong with left side recognizing part?

**Reply by Ryo624** [reply]
What happens if I re-solder the left and right TRRS connectors?
Try to re-solder it so that it blends in with the pads on the board and becomes mountainous.

The soldering on the back of the ProMicro is done nicely, so let me check the soldering of the ProMicro body on the front side.

**Reply by ricelovinbear** [reply]
I have found the reason... i'm very embarassed to say but it was that left side diodes were soldered in reverse....    I must have mistake laying out the diodes at first...
thank you so much for your time Ryo624 and sorry for confusion and thank you very much for your help!!

**Reply by Ryo624** [reply]
I'm glad you found the cause.
Please contact me if you need anything else.
